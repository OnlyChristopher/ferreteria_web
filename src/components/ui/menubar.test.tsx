import { describe, it, expect } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import {
  Menubar,
  MenubarMenu,
  MenubarTrigger,
  MenubarContent,
  MenubarItem,
  MenubarGroup,
  MenubarSeparator,
  MenubarLabel,
  MenubarCheckboxItem,
  MenubarRadioGroup,
  MenubarRadioItem,
  MenubarSub,
  MenubarSubTrigger,
  MenubarSubContent,
} from './menubar';

// Test básico de renderizado

describe('Menubar', () => {
  it('renderiza Menubar y sus subcomponentes', () => {
    render(
      <Menubar>
        <MenubarMenu>
          <MenubarTrigger>Archivo</MenubarTrigger>
          <MenubarContent>
            <MenubarItem>Nuevo</MenubarItem>
            <MenubarItem>Guardar</MenubarItem>
            <MenubarSeparator />
            <MenubarGroup>
              <MenubarLabel>Opciones</MenubarLabel>
              <MenubarCheckboxItem checked>Activado</MenubarCheckboxItem>
              <MenubarRadioGroup>
                <MenubarRadioItem value="a">A</MenubarRadioItem>
                <MenubarRadioItem value="b">B</MenubarRadioItem>
              </MenubarRadioGroup>
            </MenubarGroup>
            <MenubarSub>
              <MenubarSubTrigger>Más</MenubarSubTrigger>
              <MenubarSubContent>
                <MenubarItem>Opción extra</MenubarItem>
              </MenubarSubContent>
            </MenubarSub>
          </MenubarContent>
        </MenubarMenu>
      </Menubar>
    );
    expect(screen.getByText('Archivo')).toBeInTheDocument();
    expect(screen.getByText('Nuevo')).toBeInTheDocument();
    expect(screen.getByText('Guardar')).toBeInTheDocument();
    expect(screen.getByText('Opciones')).toBeInTheDocument();
    expect(screen.getByText('Activado')).toBeInTheDocument();
    expect(screen.getByText('A')).toBeInTheDocument();
    expect(screen.getByText('B')).toBeInTheDocument();
    expect(screen.getByText('Más')).toBeInTheDocument();
    expect(screen.getByText('Opción extra')).toBeInTheDocument();
  });

  it('abre y cierra el MenubarContent al hacer click en el trigger', () => {
    render(
      <Menubar>
        <MenubarMenu>
          <MenubarTrigger>Archivo</MenubarTrigger>
          <MenubarContent>
            <MenubarItem>Nuevo</MenubarItem>
          </MenubarContent>
        </MenubarMenu>
      </Menubar>
    );
    const trigger = screen.getByText('Archivo');
    fireEvent.click(trigger);
    expect(screen.getByText('Nuevo')).toBeInTheDocument();
  });

  it('snapshot del Menubar', () => {
    const { container } = render(
      <Menubar>
        <MenubarMenu>
          <MenubarTrigger>Archivo</MenubarTrigger>
          <MenubarContent>
            <MenubarItem>Nuevo</MenubarItem>
          </MenubarContent>
        </MenubarMenu>
      </Menubar>
    );
    expect(container).toMatchSnapshot();
  });
});

