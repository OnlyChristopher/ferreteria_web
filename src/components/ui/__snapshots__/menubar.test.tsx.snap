// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Menubar > snapshot del Menubar 1`] = `
<div>
  <div
    class="bg-background flex h-9 items-center gap-1 rounded-md border p-1 shadow-xs"
    data-orientation="horizontal"
    data-slot="menubar"
    role="menubar"
    style="outline: none;"
    tabindex="0"
  >
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="focus:bg-accent focus:text-accent-foreground data-[state=open]:bg-accent data-[state=open]:text-accent-foreground flex items-center rounded-sm px-2 py-1 text-sm font-medium outline-hidden select-none"
      data-orientation="horizontal"
      data-radix-collection-item=""
      data-slot="menubar-trigger"
      data-state="closed"
      id="radix-:r9:"
      role="menuitem"
      tabindex="-1"
      type="button"
    >
      Archivo
    </button>
  </div>
</div>
`;
